scripts/news/__main__.py

```diff
@@ -0,0 +1,243 @@
+import os
+import subprocess
+from collections import defaultdict
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Optional
+
+import click
+import requests
+
+from . import ERROR_MSG_PREFIX, __version__
+from .utils import (
+    NotRequiredIf,
+    err,
+    get_metadata_from_file,
+    get_project_meta,
+    glob_fragments,
+    load_toml_config,
+    nonceify,
+    out,
+    render_fragments,
+)
+
+
+PR_ENDPOINT = "https://api.github.com/repos/discord-modmail/modmail/pulls/{number}"
+BAD_RESPONSE = {
+    404: "Pull request not located! Please enter a valid number!",
+    403: "Rate limit has been hit! Please try again later!",
+}
+
+TEMPLATE = """
+# Please write your news content. When finished, save the file.
+# In order to abort, exit without saving.
+# Lines starting with \"#\" are ignored.
+
+""".lstrip()
+NO_NEWS_PATH_ERROR = (
+    f"{ERROR_MSG_PREFIX} `news/next/` doesn't exist.\nYou are either in the wrong directory while"
+    " running this command (should be in the project root) or the path doesn't exist, if it "
+    "doesn't exist please create it and run this command again :) Happy change-logging!"
+)
+
+CONFIG = load_toml_config()
+SECTIONS = [_type for _type, _ in CONFIG.get("types").items()]
+
+
+def save_news_fragment(ctx: click.Context, gh_pr: int, nonce: str, news_entry: str, news_type: str) -> None:
+    """Save received changelog data to a news file."""
+    date = datetime.now(timezone.utc).strftime("%Y-%m-%d")
+    path = Path(Path.cwd(), f"news/next/{news_type}/{date}.pr-{gh_pr}.{nonce}.md")
```

Comment:
The date helps order the changelogs in the directory, we are using the same order as the directory for building our changelog, so to keep it accurate we need the dates.

> Nonce would just be incremented starting at 0 if the file already exists.
Arguably, it would be nice to exclude the nonce and delimit afterwards unless the file already exists.


I am not sure I understand this correctly.